### PR TITLE
Merge pull request #6 from Crstt/main

### PR DESCRIPTION
**Pull request by Teixeira-David**
Update 11/28

### CODE REVIEW

File: ClassCompass/Assignment.swift [hunk 1]
```diff
@@ -10,12 +10,12 @@ import Foundation
 class Assignment {
     let id: Int
     let name: String
-    let dueDate: Date
+    let dueDate: Date?
```
Comment (Crstt) [inline ClassCompass/Assignment.swift:13]: Why is dueDate optional?
I think we should add a new parameter that contains the dueOnDate. This would be the information that the user fills in for when they want to do the activity. That can be optional.

File: ClassCompass/Assignment.swift [hunk 2]
```diff
@@ -10,12 +10,12 @@ import Foundation
...
     let description: String
     //points_possible
     var grade: Double?
 
-    init(id: Int, name: String, dueDate: Date, description: String, grade: Double?) {
+    init(id: Int, name: String, dueDate: Date?, description: String, grade: Double?) {
```
Comment (Crstt) [inline ClassCompass/Assignment.swift:18]: Same as above

File: ClassCompass/CanvasAPIClient.swift [hunk 3]
```diff
@@ -120,16 +123,35 @@ class CanvasAPIClient {
...
+                for assignmentDict in assignmentRaw {
+                    guard let id = assignmentDict["id"] as? Int else {
+                        print("Error: 'id' is not an Int or is missing in assignment: \(assignmentDict)")
+                        continue
+                    }
+                    guard let name = assignmentDict["name"] as? String else {
+                        print("Error: 'name' is not a String or is missing in assignment: \(assignmentDict)")
+                        continue
+                    }
+                    guard let description = assignmentDict["description"] as? String else {
```
Comment (Crstt) [inline ClassCompass/CanvasAPIClient.swift:136]: I get an error on some descriptions. For now, I would just assign "" and comment the actual value. I need to figure out why

File: ClassCompass/ccDB 2.swift
Comment (Crstt) [inline ClassCompass/ccDB 2.swift:1]: Is this file necessary?

**Review by Crstt — CHANGES_REQUESTED** [review]

File: ClassCompass/Assignment.swift (hunk 1 above)
Comment (Teixeira-David) [reply ClassCompass/Assignment.swift:13]: so when checking the pull from the API, some dueDates are an address location, not a date. So when i was pulling the program would crash because it didn't identify the correct date type. So essentially it was a nil.

File: ClassCompass/ccDB 2.swift
Comment (Teixeira-David) [reply ClassCompass/ccDB 2.swift:1]: no it is not. Not sure why it was added.
